src/Etg.Yams.Core/Storage/IDeploymentRepository.cs

```diff
@@ -1,13 +1,10 @@
 ﻿using System.Threading.Tasks;
 using Etg.Yams.Application;
-using Etg.Yams.Storage.Config;
 
 namespace Etg.Yams.Storage
 {
     public interface IDeploymentRepository
```

Comment:
What about:
```csharp
public interface IDeploymentRepository : IDeploymentConfigRepository, IApplicationsRepository
```

This would allow you to keep the current Yams Api as is and avoid breaking any users (not to mention a lot of documentation to update). You can add a class to compose an IDeploymentRepository implementation from IDeploymentConfigRepository and IApplicationsRepository:

```csharp
public class DeploymentRepository
{
   public DeploymentRepository(IDeploymentConfigRepository deploymentConfigRepository, IApplicationsRepository applicationsRepository)
   {
   }
}
```

Reply:
I was thinking about introducing `IApplicationRepository` or even separate interfaces for uploading and downloading binaries. Will implement the above and await your opinion on the need of pacjage publishing through Yams.